File: 1-two-sum/1-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int,int> m;
        vector<int> v;
        
        for(int i=0;i<nums.size();i++){
                        
            int c = target - nums[i];
            if(m.find(c) != m.end()){
                v.push_back(i);
                v.push_back(m[c]);
                return v;
            }
            m[nums[i]] = i;
             
        }
        return v;
    }
};

File: 1-two-sum/1-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        ans = []
        d = {}
        n = len(nums)
        for i in range(n):
            d[nums[i]] = i

        for i in range(n):
            if (target - nums[i]) in d.keys() and i != d[(target - nums[i])]:
                ans.append(i)
                ans.append(d[target-nums[i]])
                break
        return ans
        
        